bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/SVGFileFormat.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Michael Bangas (Vector Informatik GmbH) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.internal.image;
+
+import java.io.*;
+import java.nio.charset.*;
+import java.util.*;
+
+import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.*;
+import org.eclipse.swt.internal.DPIUtil.*;
+
+/**
+ * A {@link FileFormat} implementation for handling SVG (Scalable Vector Graphics) files.
+ * <p>
+ * This class detects SVG files based on their header and uses a registered
+ * {@link SVGRasterizer} service to rasterize SVG content.
+ * </p>
+ */
+public class SVGFileFormat extends FileFormat {
+
+	/** The instance of the registered {@link SVGRasterizer}. */
+	private static final SVGRasterizer RASTERIZER = ServiceLoader.load(SVGRasterizer.class).findFirst().orElse(null);
+
+	@Override
+	boolean isFileFormat(LEDataInputStream stream) throws IOException {
+		byte[] firstBytes = new byte[5];
+		int bytesRead = stream.read(firstBytes);
+		stream.unread(firstBytes);
+		String header = new String(firstBytes, 0, bytesRead, StandardCharsets.UTF_8).trim();
+		return header.startsWith("<?xml") || header.startsWith("<svg");
+	}
+
+	@Override
+	List<ElementAtZoom<ImageData>> loadFromByteStream(int fileZoom, int targetZoom) {
+		return Arrays.stream(loadFromByteStream(100 * targetZoom / fileZoom)).map(d -> new ElementAtZoom<>(d, targetZoom)).toList();
```

Comment:
I ask in another comment for whether the rasterizer cannot just return a single ImageData. In that case, I would merge the two `loadFromByteStream` methods and just do the transformation to a singleton list with one ElementAtZoom directly in that method.

Reply:
I merged the two methods and changed `Arrays.stream().map().toList()` to `List.of()` as now the Rasterizer does  directly return `ImageData` instead of `ImageData[]`. Way cleaner now :).